code/game/objects/items/weapons/teleprod.dm

```diff
@@ -0,0 +1,50 @@
+/obj/item/weapon/melee/baton/cattleprod/teleprod
+	name = "teleprod"
+	desc = "A prod with a bluespace crystal on the end. The crystal doesn't look too fun to touch."
+	icon_state = "teleprod_nocell"
+	item_state = "teleprod"
+	origin_tech = "combat=2;bluespace=4;materials=3"
+
+/obj/item/weapon/melee/baton/cattleprod/teleprod/attack(mob/living/carbon/M, mob/living/carbon/user)//handles making things teleport when hit
+	..()
+	if(status && user.disabilities & CLUMSY && prob(50))
+		user.visible_message("<span class='danger'>[user] accidentally hits themself with [src]!</span>", \
+							"<span class='userdanger'>You accidentally hit yourself with [src]!</span>")
+		user.Weaken(stunforce*3)
+		deductcharge(hitcost)
+		do_teleport(user, get_turf(user), 50)//honk honk
+		return
+	else
+		if(status)
+			if(!istype(M) && M.anchored)
```

Comment:
okay so after fixing teleprods and testing again, i can confirm the anchor check works. if a mob is buckled (anchored) and hit with a teleprod, they automatically unbuckle, meaning there's no problemo.

the only real issue i found was that if you're clumsy and accidentally hit yourself when trying to hit someone else, they still teleport even though they're not stunned. not sure how to fix this one.